quarkus/config-api/src/main/java/org/keycloak/config/CachingOptions.java

```diff
@@ -40,4 +53,83 @@ public enum Stack {
+            .description("The TLS provider. It defaults to the JDK implementation. Other providers may require adding libs/jars to the classpath.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_KEYSTORE = new OptionBuilder<>(JGROUPS_TLS_KEYSTORE_FILE_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The Keystore file path. The Keystore must contain the certificate to use by the TLS protocol. " +
+                    "By default, it lookup 'jgroups-keystore.p12' under conf/ directory.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_KEYSTORE_PASSWORD = new OptionBuilder<>(JGROUPS_TLS_KEYSTORE_PASSWORD_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The password to access the Keystore.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_KEYSTORE_TYPE = new OptionBuilder<>(JGROUPS_TLS_KEYSTORE_TYPE_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The Keystore format. It defaults to PKCS#12 format.")
+            .defaultValue("pkcs12")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_KEYSTORE_ALIAS = new OptionBuilder<>(JGROUPS_TLS_KEYSTORE_ALIAS_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The certificate alias in the Keystore.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_TRUSTSTORE = new OptionBuilder<>(JGROUPS_TLS_TRUSTSTORE_FILE_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The Truststore file path. " +
+                    "It should contain the trusted certificates or the Certificate Authority that signed the certificates. " +
+                    "By default, it lookup 'jgroups-truststore.p12' under conf/ directory.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_TRUSTSTORE_PASSWORD = new OptionBuilder<>(JGROUPS_TLS_TRUSTSTORE_PASSWORD_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The password to access the Truststore.")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_TRUSTSTORE_TYPE = new OptionBuilder<>(JGROUPS_TLS_TRUSTSTORE_TYPE_PROPERTY, String.class)
+            .category(OptionCategory.CACHE)
+            .description("The Truststore format. It defaults to PKCS#12 format.")
+            .defaultValue("pkcs12")
+            .buildTime(true)
+            .build();
+
+    public static final Option<String> JGROUPS_TLS_CLIENT_AUTH = new OptionBuilder<>(JGROUPS_TLS_CLIENT_AUTH_PROPERTY, String.class)
```

Comment:
From what we discussed earlier we want to prevent unauthorized clients to join the communication. So the only option that makes sense IMHO is `need` here. If anything else is not a common use case, we should remove this option and hard-code it to `need`. A user of Keycloak can still change the XML if they want to do something similar. 

If we allow for this option to be changed, I assume people will choose an unsecure setting by accident.

Reply:
Makes sense.